### PR TITLE
Clean Arch with some tests

### PR DESCRIPTION
**Pull request by kanake10**
This is an enhancement PR, which does the following:

- moves the repositoryImpl classes from domain package to data to follow clean Architecture
- correctly renames vals in dto to match those in domain to make sure tests pass
- ensure correct package importation 
- create a separate mappers file to easily test it
- Tests the DtoToDomain file,passing
- debugs the api calls using okhttp
- Runs a CI pipeline using github actions

Tests Passing
Screenshot 2022-11-15 at 00 36 36

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        run: ./gradlew build
+
+      - name: Run Tests
+        run: ./gradlew test
+
+      - name: Upload a Build Artifact (APK)
+        uses: actions/upload-artifact@v2.2.4
+        with:
+          name: app
+          path: app/build/outputs/apk/debug/app-debug.apk
```
Comment (otsembo) [inline .github/workflows/build.yml:31]: Good Job on setting up the CI.

File: app/src/test/java/com/eeyan/cryptapp/data/mappers/DtoToDomainKtTest.kt [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    fun `confirm that CoinDto toCoin results to Coin`() {
+        val coinDto = CoinDto(
+            id = "android",
+            is_active = false,
+            is_new = true,
+            name = "name",
+            rank = 10,
+            symbol = "symbol",
+            type = "type"
+        )
```
Comment (otsembo) [inline app/src/test/java/com/eeyan/cryptapp/data/mappers/DtoToDomainKtTest.kt]: To shorten the number of lines in the test class, you could also shift the constants to TestUtils

**Review by otsembo — APPROVED** [review]
Godd Job on the PR @kanake10.
You can try refactoring the test class and then close the PR.

File: app/src/test/java/com/eeyan/cryptapp/data/mappers/DtoToDomainKtTest.kt (hunk 2 above)
Comment (kanake10) [reply app/src/test/java/com/eeyan/cryptapp/data/mappers/DtoToDomainKtTest.kt]: Working on it ✊

File: .github/workflows/build.yml (hunk 1 above)
Comment (kanake10) [reply .github/workflows/build.yml:31]: ✊

**Pull request merged**